lib/node_modules/@stdlib/math/generics/random/sample/lib/factory.js

```diff
@@ -20,35 +25,119 @@ var slice = Array.prototype.slice;
 /**
 * Returns a function to sample elements from an array-like object.
 *
-* @param {PositiveInteger} seed - integer-valued seed
+* @param {ArrayLike} [pool] - array-like object to sample from
+* @param {Options} [options] - function options
+* @param {PositiveInteger} [options.seed] - integer-valued seed
+* @param {NonNegativeInteger} [options.size] - sample size
+* @param {boolean} [options.replace=true] - boolean indicating whether to sample with replacement
 * @returns {Function} function to sample elements from an array-like object
 *
 * @example
-* var sample = factory( 323 );
+* var sample = factory({
+*     'seed': 232
+* });
+* var out = sample( 'abcdefg' );
+* // [ 'g', 'd', 'g', 'f', 'c', 'e', 'f' ]
+*
+* @example
+* var sample = factory( [1,2,3,4,5,6], {
+*     'seed': 232
+*     'size': 2
+* });
+* var out = sample();
+* // returns [ 6, 4 ]
+*
+* out = sample()
+* // returns [ 6, 5 ]
+*
+* @example
+* var sample = factory( [1,2,3,4,5,6], {
+*     'seed': 474
+*     'size': 3
+*     'replace': false
+* });
+* var out = sample();
+* // returns [ 4, 3, 6 ]
+*
+* out = sample()
+* // returns [ 1, 5, 2 ]
+*
+* out = sample()
+* // throws <Error>
+*
+* @example
+* var sample = factory( [0,1], {
+*     'size': 2
+* });
+*
+* var out = sample();
+* // e.g., returns [ 1, 1 ]
+*
+* out = sample({
+*     size: 10
+* });
+* // e.g, returns [ 0, 1, 1, 1, 0, 1, 0, 0, 1, 1 ]
+*
+* @example
+* var sample = factory( [0,1], {
+*     'size': 2
+* });
+*
+* var out = sample();
+* // e.g., returns [ 1, 1 ]
+*
+* out = sample({
+*     'replace': false
+*     'size': 1
+* });
+* // returns [ 0 ] or [ 1 ]
+*
+* out = sample();
+* // throws <RangeError>
```

Comment:
Hmm. What happens right now is that `sample` will not throw, but returns `[1,1]` in case `[0]` was drawn before and `[0,0]` if `[1]` was drawn in the previous call. So disregard the `RangeError`, that is not up-to-date. However, I am wondering why you would expect to never receive `null` when specifying `replace=true` as the default. If you specify `replace=false` in even one call, what happens right now is that balls are taken out of the `pool` urn without putting them back in afterwards, so this will affect all subsequent calls.